sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -31,7 +31,7 @@ public static BinaryData Write<T>(T model, ModelReaderWriterOptions? options = d
 
         options ??= ModelReaderWriterOptions.Json;
 
-        if (IsJsonFormatRequested(model, options) && model is IJsonModel<T> jsonModel)
+        if (ShouldWriteAsJson(model, options, out IJsonModel<T>? jsonModel))
```

Comment:
There are a few instances of this if state in this file can we update each of them to use the new helper?